File: java/src/main/java/fr/sdecout/handson/rest/library/LibrarySearch.java
package fr.sdecout.handson.rest.library;

import fr.sdecout.handson.rest.author.AuthorId;
import fr.sdecout.handson.rest.book.Isbn;

import java.util.stream.Stream;

public interface LibrarySearch {
    Stream<LibrarySearchResponseItem> searchLibrariesClosestTo(PostalCode postalCode);
    Stream<LibrarySearchResponseItem> searchLibrariesWithBookAvailable(Isbn isbn);
    Long countLibrariesWithBooksBy(AuthorId author);
}

File: kotlin/src/main/kotlin/fr/sdecout/handson/rest/library/LibrarySearch.kt
package fr.sdecout.handson.rest.library

import fr.sdecout.handson.rest.author.AuthorId
import fr.sdecout.handson.rest.book.Isbn

interface LibrarySearch {
    fun searchLibrariesClosestTo(postalCode: PostalCode): List<LibrarySearchResponseItem>
    fun searchLibrariesWithBookAvailable(isbn: Isbn): List<LibrarySearchResponseItem>
    fun countLibrariesWithBooksBy(author: AuthorId): Long
}
